README.md

```diff
@@ -8,16 +8,165 @@ Local.
+<!-- TOC -->
+
+- [loopback-connector-cloudant](#loopback-connector-cloudant)
+- [Design](#design)
+    - [No partial update](#no-partial-update)
+    - [No frequent modification](#no-frequent-modification)
+    - [Conflict control](#conflict-control)
+- [Model](#model)
+    - [Map between model and document](#map-between-model-and-document)
+    - [Model-specific configuration](#model-specific-configuration)
+    - [_rev property](#_rev-property)
+- [Installation](#installation)
+- [Config](#config)
+    - [generate datasource](#generate-datasource)
+    - [datasource config(`url` vs `username` & `password`)](#datasource-configurl-vs-username--password)
+    - [requests plugin](#requests-plugin)
+    - [example usage](#example-usage)
+- [CRUD](#crud)
+    - [update](#update)
+- [Migration](#migration)
+    - [autoupdate vs automigrate](#autoupdate-vs-automigrate)
+    - [isActual](#isactual)
+    - [example code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Setup Cloudant instance](#setup-cloudant-instance)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Design
+Loopback tries best to fit its model to a specific database's design, while limited by the nature of database, it's not always possible to support all loopback features perfectly, and user would like to be aware of some key features about cloudant before they start to design a cloudant model.
+
+## No partial update
+
+*Cloudant does not support the idea of updating a document. All "updates" on a document are _destructive_ replacements.*
 
-## Key features
+For example:
+
+```
+// original document
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "1",
+  "prop2": "2",
+}
+
+// data to be updated
+ds.updateOrCreate('User', {
+  prop1: 'updated1',
+}, function (err, res) {});
+
+// document after update
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "updated1",
+}
+```
+
+*Please note how property `prop2` was completely dropped upon update.*
+
+Workaround: Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
+
+We plan to support partial update in the future by retrieving the original data and merging it with the updated one, but user should be aware of the risk when merging happens.
+
+## No frequent modification
+
+Cloudant is not designed to change same document frequently and multiple times. It stores status change by creating different documents and including the same unique id to tell that they are attached to the same item, not updating the same document.
+
+By modeling the data in separate documents that are only written once, we can ensure that documents never become conflicted, such as during concurrent access to the same document by separate processes.
+
```

Comment:
It implicates user need to properly handle doc conflicts by dealing with its revision. I will add that part.